docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -901,6 +901,42 @@ FROM rides
 GROUP BY 1
 ```
 
+### @DEDUPLICATE
+
+`@DEDUPLICATE` is used to deduplicate rows in a table based on the specified partition and order columns with a window function.
+
+It supports the following arguments, in this order:
+
+- `relation`: The table or CTE name to deduplicate
+- `partition_by`: column names, or expressions to use to identify a window of rows out of which to select one as the deduplicated row
+- `order_by`: A list of strings representing the ORDER BY clause
+
+For example, the following query:
+```sql linenums="1"
+with raw_data as (
+@deduplicate(my_table, [id, cast(event_date as date)], ['event_date DESC', 'status ASC'])
+)
+
+select * from raw_data
+```
+
+would be rendered as:
+
+```sql linenums="1"
+WITH "raw_data" AS (                                                                           
+  SELECT                                                                                       
+    *                                                                                          
+  FROM "my_table" AS "my_table"                                                                
+  QUALIFY                                                                                      
+    ROW_NUMBER() OVER (PARTITION BY "id", CAST("event_date" AS DATE) ORDER BY "event_date"     
+DESC, "status" ASC) = 1                                                                        
```

Comment:
nit: is this the actual render output? Formatting looks a bit weird.

Reply:
yes it does when I render an example sql model.
![image](https://github.com/user-attachments/assets/2c4678d0-787d-4bff-b0a7-6d459b13d84a)

Reply:
Perhaps it happens due to the terminal width. To be clear, I'm talking about the `"event_date" DESC, 1223) = 1` being in a separate line.

Reply:
ahh yes, it's due to terminal width. here it is with a wider terminal

![image](https://github.com/user-attachments/assets/d2535b22-db6f-4c8c-acbb-9172d7c15db3)

Reply:
I'll update the example in this markdown so people don't get confused with line wrapping.

Reply:
sgtm 👍